node/core/approval-voting/src/lib.rs

```diff
@@ -768,25 +866,50 @@ async fn handle_actions(
 				let block_hash = indirect_cert.block_hash;
 				let validator_index = indirect_cert.validator;
 
-				ctx.send_unbounded_message(ApprovalDistributionMessage::DistributeAssignment(
-					indirect_cert,
-					candidate_index,
-				).into());
+				match currently_checking_set.approvals_cache.get(&candidate_hash) {
+					Some(ApprovalOutcome::Approved) => {
+						let mut sender = ctx.sender().clone();
+						let _ = issue_approval(
+							&mut sender,
+							state,
+							metrics,
+							ApprovalVoteRequest {
+								validator_index,
+								block_hash,
+								candidate_index,
+							}
+						)?;
 
-				let handle = launch_approval(
-					ctx,
-					metrics.clone(),
-					background_tx.clone(),
-					session,
-					&candidate,
-					validator_index,
-					block_hash,
-					candidate_index as _,
-					backing_group,
-				).await?;
+						let _ = currently_checking_set.insert_relay_block_hash(
```

Comment:
Why are we adding this to the currently_checking_set? Anything in the cache should be beyond the point of 'currently checking'.

Reply:
damn, this should have been caught by me. Making a note of it.